lib/share_progress/button.rb

```diff
@@ -4,24 +4,39 @@
 module ShareProgress
   class Button
 
+    attr_accessor :page_url, :page_title, :button_template, :share_button_html, :is_active
+    attr_reader   :id
+
     class << self
-      def create
+
+      def create(page_url, button_template, raw_options={})
+        created = update(nil, page_url, button_template, raw_options)
+        created.nil? ? nil : new(created)
       end
 
-      def update
+      # this method is used by instance.save and Button.create
+      def update(id, page_url, button_template, raw_options={})
+        options = filter_keys(raw_options, optional_keys)
+        options[:advanced_options] = filter_keys(options[:advanced_options], advanced_options_keys)
+        options = options.merge({page_url: page_url, button_template: button_template})
+        options[:id] = id unless id.nil? # without ID, update is create
+        created = Client.post endpoint('update'), { body: options }
+        created[0]
       end
 
       def find(id)
-        Client.get endpoint('read'), { query: { id: id } }
+        matches = Client.get endpoint('read'), { query: { id: id } }
+        raise ArgumentError.new("No button exists with id #{id}") if matches.size < 1
```

Comment:
@NealJMD good call bringing this up in chat. 

This isn't really an `ArgumentError` - but nor are there any core ruby errors that are better suited. 

It's typical to create your own custom errors for the domain at hand:

```ruby
# lib/error.rb
module ShareProgress
  # for rescuing from all share progress errors.
  class ShareProgressError < StandardError; end

  class NotFound < ShareProgressError; end
end
```

Now you can raise a `NotFound` error, which is much more descriptive.